services-api/src/main/java/io/scalecube/services/ExtendedMicroservicesContext.java

```diff
@@ -0,0 +1,21 @@
+package io.scalecube.services;
+
+import io.scalecube.net.Address;
+import io.scalecube.services.discovery.api.ServiceDiscovery;
+
+/**
+ * Context of Scale Cube node. Used in {@link ServiceFactory}. Compared to MicroservicesContext, it
+ * gives access to ServiceDiscovery.
+ *
+ * @see ServiceFactory
+ */
+public interface ExtendedMicroservicesContext extends MicroservicesContext {
+
+  /**
+   * Service discovery for services localed in other nodes.
+   *
+   * @return service discovery
+   * @see ServiceDiscovery
+   */
+  ServiceDiscovery serviceDiscovery();
```

Comment:
Move it to `MicroserviceContext`

Reply:
ServiceFactory#getServiceDefinitions consume MicroserviceContext, but MicroserviceContext#serviceDiscovery return null

Reply:
Until u read all service-definitions dsicovery will not be ready for you. It's a contract.